packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -499,7 +499,7 @@ export const coreDefinitions: {[coreSettingName: string]: SettingsDefinition} =
   checksumBehavior: {
     description: `Enumeration defining what to do when a checksum doesn't match expectations`,
     type: SettingsType.STRING,
-    default: `throw`,
+    default: isCI ? `reset` : `throw`,
```

Comment:
Why that?

Reply:
I'm not sure why, but when I've changed:
1) linker from nm to pnp
2) compression level from default to 0
3) upgraded yarn
on CI I've got checksum mismatch error. Cache was from previous ci runs and is shared.
And then I thought, from design perspective, maybe default behaviour on CI should be to override existing cache. 

But, maybe rabbit hole is deeper, and if think about two  simultaneous CI runs which using same cache, have same locators but checksum are different. 
Second yarn install will override file from first install.

Reply:
We can't change this, if the user is using Zero-Installs it must throw when it doesn't match so the user is made aware of the corrupted cache so they can fix it locally.

> on CI I've got checksum mismatch error. Cache was from previous ci runs and is shared

That shouldn't happen, not due to the steps you listed anyways.

Reply:
reverted

Reply:
> That shouldn't happen, not due to the steps you listed anyways.

Yeah, this seems like a bug with patch protocol + compression level change. Will investigate it.